staging/src/k8s.io/apiserver/pkg/storage/etcd3/preflight/checks.go

```diff
@@ -59,9 +60,12 @@ func parseServerURI(serverURI string) (*url.URL, error) {
 // CheckEtcdServers will attempt to reach all etcd servers once. If any
 // can be reached, return true.
 func (con EtcdConnection) CheckEtcdServers() (done bool, err error) {
-	// Attempt to reach every Etcd server in order
-	for _, serverURI := range con.ServerList {
-		host, err := parseServerURI(serverURI)
+	rand.Seed(time.Now().UTC().UnixNano())
+	// Attempt to reach every Etcd server randomly.
```

Comment:
how does this work with etcd2 support ?

Reply:
It serves for both etcd2 etcd3.

Reply:
odd to see the random number generator in a func.

Reply:
ok, i will remove it.

Reply:
done